lib/index.js

```diff
@@ -140,6 +140,14 @@ function getOptions(options) {
   options.sourceMap = getSourceMap(options);
   options.style = getStyle(options) || 0;
 
+  // context object represents node-sass environment
+  options.context = {
+    info: module.exports.info,
+    state: {},
+    sync: false,
```

Comment:
@Jakobo, generally these are very good ideas. But after few iterations, you will notice that new use-cases will emerge for the features we initially took lightly. As soon as you try to do something creative here, it may cause chaos in the, the tertiary, quaternary .. downstream of node-sass.

Therefore, I think it is a good idea to keep the core API side as simple/dumb as possible unless of course it is utterly necessary. 

For example, we are even resolving the output/source-map file paths just because libsass require it. And we will be removing the (remaining) default values for options and let libsass set them (https://github.com/sass/node-sass/issues/677).

For this PR. I propose to let the consumers:

* use or misuse `options` object and realise if something goes wrong while misusing it. :)
* clone the `options` object before passing it to node-sass if they are sharing it elsewhere for other purposes.
* figure out whether the custom function or importer is sync or async if necessary. If they are really defining a function to work with both render[Sync], they can just `return` the value and it will work for both. For more mysterious use-case (callback for async and return for sync), they are free to override `this.options` with custom payload.
* make use of `this.options` in importers, success and error callbacks.